homeassistant/components/synology_dsm/__init__.py

```diff
@@ -250,6 +261,11 @@ async def async_setup(self):
             device_token=self._entry.data.get("device_token"),
         )
 
+        await self._hass.async_add_executor_job(self.dsm.discover_apis)
```

Comment:
Should be done automatically, needed ?

Reply:
I want to know if the camera API existed in the next line, so this is actually needed.

Reply:
Oh yep!

Because we don't call `dsm.login()` !

(same for `information` not retreive at beginning).

I will change it later, when fetching with `DataUpdateCoordinator`